Feature controller gamelogic tournament and tournamentsettings

RobAnKo opened the pull request:
added Controller folder
Introduced GameLogicTournament and GameLogicTournamentSettings (stubs)
added some basic functions

Commit:
introduced BasicFunctions, GameLogicTournament and GameLogicTournamentSettings

Commit:
added Controller folder
Introduced GameLogicTournament and GameLogicTournamentSettings (stubs)
added some basic functions

sanchez-chris commented on TiKiTuTo/Controller/BasicFunctions.cs:
```diff
@@ -0,0 +1,128 @@
+
+
+
+        //TODO
```
nichts implementiert (?)

RobAnKo replied:
Ja sorry bro ich war muede. Ich arbeite morgen daran und pushe erneut mit implementierung.

Commit:
removed comment (test for PR)

Commit:
reconfigured structure:
BasicFunctions empty -> methods moved to Inputhandler and View,
Getting Input now through InputHandler,
View now does ReadInput(),
Controller has a dummy StartTournament method

Commit:
created references to the view in methods in InputHandler and GameLogic methods. moved CreateTeam() and CreateNTeams() into BasicFunctions
added Model namespace reference in JSONService
adjusted program for a minimum working solution in creating a new tournament

RobAnKo commented:
Hey Leute!

Ich habe mit den letzten zwei commits die angedachte Struktur umgesetzt. Folgende key changes:
- Methoden-Calls von InputHandler und GameLogic bekommen jetzt immer den View als Argument mit. Damit können sie statisch bleiben.
- Der InputHandler implementiert alle Methoden, bei denen es darum geht einen Input vom User abzufragen
- Der InputValidator testet für verschiedene Werte, ob sie gültig sind und gibt booleans aus.
- momentan haben wir folgende Namespaces: Tikituto.Model, Tikituto.View, Tikituto.Controller

Um das Ganze zum Laufen zu bringen, musste ich auch paar kleine Änderungen an program.cs, JSONService.cs und anderen Dateien machen, die eigtl nicht mein Zuständigkeitsbereich sind. 
Schaut euch gerne über die "Files changed"-Reiter (so heißts auf Englisch) oben rechts die Änderungen an. 

Ideen von meiner Seite: Entweder ihr schaut euch den PR an und kommentiert eure Gedanken/Fragen über die Github-Funktionen ODER wir checken Montag zusammen den PR ODER eine Kombination daraus.

In jedem Fall: Sagt was euch stinkt, was ihr nicht nachvollziehen könnt, was zu viel und/oder zu wenig ist. 

PS: 
- InputValidator-Methoden größtenteils Platzhalter (Implementierung Mahdis Job)
- Methodenbeschreibungen (///) kommen in einem nächsten Commit.

Commit:
added XML documentation comments

sanchez-chris commented on TiKiTuTo/Controller/BasicFunctions.cs:
```diff
@@ -0,0 +1,50 @@
+
+
+
+        public static List<Team> CreateNTeams(int NumberOfTeamsTotal, IView View)
```
wieso CreateNTeams()? wäre nicht deutlicher CreateListOfTeams() oder ein ähnlicher Name??

RobAnKo replied:
Guter Punkt, werde ich anpassen.

sanchez-chris commented on TiKiTuTo/Controller/Controller.cs:
```diff
@@ -0,0 +1,26 @@
+using TiKiTuTo.View;
+using TiKiTuTo.Model;
+
+namespace TiKiTuTo.Controller
```
ich habe immer TikiTuTo.Controller etwas redundant gefunden. Wäre meine Meinung nach besser nur Controller. Auch nicht so relevant. Oder gibt es einen Grund, deshalb es so heißen soll?

ForestOfEntropy replied:
Das Projekt heißt ja Tikituto, und der Controller is ja in der Struktur unter dem Projekt angesiedelt und kann so auf die Validator und Helferklassen zugreifen. (Ist auch einheitlicher vom Naming her, also Tikituto.View/.Controller/.Model)

RobAnKo replied:
Ich bin hier auch Chris' Meinung, da alles im Tikituto-Namespace passiert können wir auch einfach Controller/Model/View nutzen, ganz ohne "Tikituto."

ForestOfEntropy replied:
Wenn das in C# keine Probleme macht können wir das meiner Meinung nach gerne ändern.

DominikSpeidel replied:
okay gerne ändern

sanchez-chris commented on TiKiTuTo/Controller/InputValidator.cs:
```diff
@@ -0,0 +1,56 @@
+    static class InputValidator
+    {
+
+        //static bool ValidateTournamentSettings(TournamentSettings)
```
warum ist es kommentiert? nicht aktuell? nicht richtig? noch zu tun?

RobAnKo replied:
Nicht 100% sicher, ob wir es brauchen. Auch noch nicht implementiert -> TODO. Ich habe ein paar noch nicht implementierte Methoden hier und da, damit wir jetzt erstmal die generelle Struktur haben und damit auch andere Menschen noch implementierung schreiben können:)

sanchez-chris commented on TiKiTuTo/Controller/InputValidator.cs:
```diff
@@ -0,0 +1,56 @@
+
+        }
+
+        public static bool IsValidNumberOfTeamsInKORound(int NumberInKO, int NumberTeamsTotal)
```
ist das nicht immer false? Warum macht man nichts mit den Argumente NumberInKO und NumberTeamsTotal?

RobAnKo replied:
Noch nicht implementiert -> TODO -> Inputvalidator ist laut Plan Mahdis Task.

sanchez-chris replied:
alles klar

sanchez-chris commented on TiKiTuTo/Controller/InputValidator.cs:
```diff
@@ -0,0 +1,56 @@
+
+        }
+
+        public static bool IsValidNumberOfPreliminaryGamesPerTeam(int GamesPerTeam, int NumberTeamsTotal)
```
same here

ForestOfEntropy replied:
>warum ist es kommentiert? nicht aktuell? nicht richtig? noch zu tun?
>ist das nicht immer false? Warum macht man nichts mit den Argumente >NumberInKO und NumberTeamsTotal?
> same here


Die mache ich, Robin hat nur die Struktur aufgebaut, damit es einheitlicher ist

sanchez-chris replied:
alles klar

sanchez-chris commented:
hi @RobAnKo ich habe ein Paar Comments hinzugefügt.
